src/extra/components/scan.py

```diff
@@ -163,6 +162,83 @@ def plot(self, ax=None):
 
         return ax
 
+    def bin(self, data):
```

Comment:
I would prefer a more expressive name here, as it initially confused me what is actually binned here to what. On top of that, I'm not a fan of collisions with builtins.

As I understand, this method can take arbitrary train-resolved data and averages it for each bin this scan represents. Binning in itself doesn't mean averaging to me, it just means combining results based on some metric. What do you think of explicitly calling it `.step_averages()`?

Reply:
Sure, renamed it to `step_average()` in dd35a73 (`average_by_steps()` is a bit long methinks).